Ensure manual unit tests can be rerun

If the local instance of the application creates an svr model, as it's first model the manual `pytest` unit test will fail.  Specifically, the unit test expects the first stored dataset instance to be an svm dataset.  Therefore, we may need to adjust the logic flow, by referencing another unique identifier (other than the `session_id`).

Instead of running the inherent flask `app.py`, we manually implemented our installed `gunicorn`:

```bash
vagrant@vagrant-ubuntu-trusty-64:/vagrant$ gunicorn -b 0.0.0.0:5000 'app:create_app()'
[2016-11-27 23:34:16 +0000] [3713] [INFO] Starting gunicorn 19.6.0
[2016-11-27 23:34:16 +0000] [3713] [INFO] Listening at: http://0.0.0.0:5000 (3713)
[2016-11-27 23:34:16 +0000] [3713] [INFO] Using worker: sync
[2016-11-27 23:34:16 +0000] [3718] [INFO] Booting worker with pid: 3718
```

Now, when we run our `manual` unit tests two times (back to back):

```bash
vagrant@vagrant-ubuntu-trusty-64:/vagrant/test$ pytest manual
===================================== test session starts ======================================
platform linux2 -- Python 2.7.6, pytest-3.0.4, py-1.4.31, pluggy-0.4.0
rootdir: /vagrant/test/manual, inifile: pytest.ini
plugins: flask-0.10.0
collected 20 items

manual/configure_database.py .
manual/configure_redis.py .
manual/authentication/pytest_crypto.py .
manual/authentication/pytest_validate_password.py .
manual/programmatic_interface/dataset_url/pytest_svm_dataset_url.py ....
manual/programmatic_interface/dataset_url/pytest_svr_dataset_url.py ....
manual/programmatic_interface/file_upload/pytest_svm_file_upload.py ....
manual/programmatic_interface/file_upload/pytest_svr_file_upload.py ....

================================== 20 passed in 76.13 seconds ==================================
vagrant@vagrant-ubuntu-trusty-64:/vagrant/test$ pytest manual
===================================== test session starts ======================================
platform linux2 -- Python 2.7.6, pytest-3.0.4, py-1.4.31, pluggy-0.4.0
rootdir: /vagrant/test/manual, inifile: pytest.ini
plugins: flask-0.10.0
collected 20 items

manual/configure_database.py .
manual/configure_redis.py .
manual/authentication/pytest_crypto.py .
manual/authentication/pytest_validate_password.py .
manual/programmatic_interface/dataset_url/pytest_svm_dataset_url.py ....
manual/programmatic_interface/dataset_url/pytest_svr_dataset_url.py ....
manual/programmatic_interface/file_upload/pytest_svm_file_upload.py ....
manual/programmatic_interface/file_upload/pytest_svr_file_upload.py FFFF

=========================================== FAILURES ===========================================
________________________________________ test_data_new _________________________________________

    def test_data_new():
        '''@test_data_new

        This method tests the 'data_new' session.

        '''

>       assert requests.post(
            endpoint_url,
            headers=headers,
            data=get_sample_json('svr-data-new.json', 'svr')
        )
E       assert <Response [500]>
E        +  where <Response [500]> = <function post at 0x7f5f69044c08>('http://localhost:5000/load-data', headers={'Content-Type': 'application/json'}, data='{"properties": {"model_type": "svr", "session_name": "sample_svr_title", "dataset_type": "file_upload", "session_type...-2": 95.03, "predictor-3": 0.488, "predictor-4": 295, "predictor-5": 55.83, "predictor-6": 0.001, "predictor-7": 27}}}')
E        +    where <function post at 0x7f5f69044c08> = requests.post
E        +    and   '{"properties": {"model_type": "svr", "session_name": "sample_svr_title", "dataset_type": "file_upload", "session_type...-2": 95.03, "predictor-3": 0.488, "predictor-4": 295, "predictor-5": 55.83, "predictor-6": 0.001, "predictor-7": 27}}}' = get_sample_json('svr-data-new.json', 'svr')

manual/programmatic_interface/file_upload/pytest_svr_file_upload.py:71: AssertionError
_______________________________________ test_data_append _______________________________________

    def test_data_append():
        '''@test_data_append

        This method tests the 'data_append' session.

        '''

>       assert requests.post(
            endpoint_url,
            headers=headers,
            data=get_sample_json('svr-data-append.json', 'svr')
        )
E       assert <Response [500]>
E        +  where <Response [500]> = <function post at 0x7f5f69044c08>('http://localhost:5000/load-data', headers={'Content-Type': 'application/json'}, data='{"properties": {"model_type": "svr", "dataset_type": "file_upload", "session_id": "4", "session_type": "data_append"}...-2": 92.22, "predictor-3": 0.356, "predictor-4": 235, "predictor-5": 64.45, "predictor-6": 0.001, "predictor-7": 31}}}')
E        +    where <function post at 0x7f5f69044c08> = requests.post
E        +    and   '{"properties": {"model_type": "svr", "dataset_type": "file_upload", "session_id": "4", "session_type": "data_append"}...-2": 92.22, "predictor-3": 0.356, "predictor-4": 235, "predictor-5": 64.45, "predictor-6": 0.001, "predictor-7": 31}}}' = get_sample_json('svr-data-append.json', 'svr')

manual/programmatic_interface/file_upload/pytest_svr_file_upload.py:85: AssertionError
_____________________________________ test_model_generate ______________________________________

    def test_model_generate():
        '''@test_model_generate

        This method tests the 'model_generate' session.

        '''

>       assert requests.post(
            endpoint_url,
            headers=headers,
            data=get_sample_json('svr-model-generate.json', 'svr')
        )
E       assert <Response [500]>
E        +  where <Response [500]> = <function post at 0x7f5f69044c08>('http://localhost:5000/load-data', headers={'Content-Type': 'application/json'}, data='{"properties": {"model_type": "svr", "sv_kernel_type": "poly", "session_id": "4", "session_type": "model_generate"}}')
E        +    where <function post at 0x7f5f69044c08> = requests.post
E        +    and   '{"properties": {"model_type": "svr", "sv_kernel_type": "poly", "session_id": "4", "session_type": "model_generate"}}' = get_sample_json('svr-model-generate.json', 'svr')

manual/programmatic_interface/file_upload/pytest_svr_file_upload.py:99: AssertionError
______________________________________ test_model_predict ______________________________________

    def test_model_predict():
        '''@test_model_predict

        This method tests the 'model_predict' session.

        '''

>       assert requests.post(
            endpoint_url,
            headers=headers,
            data=get_sample_json('svr-model-predict.json', 'svr')
        )
E       assert <Response [500]>
E        +  where <Response [500]> = <function post at 0x7f5f69044c08>('http://localhost:5000/load-data', headers={'Content-Type': 'application/json'}, data='{"properties": {"model_id": "4", "prediction_input[]": ["22.22", "96.24", "338", "72.55", "0.001", "28", "0.678"], "session_type": "model_predict"}}')
E        +    where <function post at 0x7f5f69044c08> = requests.post
E        +    and   '{"properties": {"model_id": "4", "prediction_input[]": ["22.22", "96.24", "338", "72.55", "0.001", "28", "0.678"], "session_type": "model_predict"}}' = get_sample_json('svr-model-predict.json', 'svr')

manual/programmatic_interface/file_upload/pytest_svr_file_upload.py:113: AssertionError
============================ 4 failed, 16 passed in 138.81 seconds =============================
```

Correspondingly, our first terminal console displays the following:

```bash
vagrant@vagrant-ubuntu-trusty-64:/vagrant$ gunicorn -b 0.0.0.0:5000 'app:create_app()'
[2016-11-27 23:34:16 +0000] [3713] [INFO] Starting gunicorn 19.6.0
[2016-11-27 23:34:16 +0000] [3713] [INFO] Listening at: http://0.0.0.0:5000 (3713)
[2016-11-27 23:34:16 +0000] [3713] [INFO] Using worker: sync
[2016-11-27 23:34:16 +0000] [3718] [INFO] Booting worker with pid: 3718
[2016-11-27 23:38:21,523] ERROR in app: Exception on /load-data [POST]
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1988, in wsgi_app
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1641, in full_dispatch_request
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1544, in handle_user_exception
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1639, in full_dispatch_request
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1625, in dispatch_request
  File "/vagrant/interface/views.py", line 99, in load_data
  File "/vagrant/brain/load_data.py", line 78, in load_data_new
  File "/vagrant/brain/session/base_data.py", line 129, in save_premodel_dataset
  File "/vagrant/brain/session/data/save_dataset.py", line 34, in dataset
  File "/vagrant/brain/database/save_feature.py", line 86, in save_feature
  File "/vagrant/brain/database/db_query.py", line 98, in sql_connect
AttributeError: 'SQL' object has no attribute 'cursor'
[2016-11-27 23:38:21 +0000] [3718] [ERROR] Error handling request /load-data
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/sync.py", line 135, in handle
  File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/sync.py", line 176, in handle_request
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 2000, in __call__
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1991, in wsgi_app
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1746, in make_response
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/wrappers.py", line 847, in force_type
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/wrappers.py", line 56, in _run_wsgi_app
ImportError: No module named test
[2016-11-27 23:38:21,577] ERROR in app: Exception on /load-data [POST]
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1988, in wsgi_app
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1641, in full_dispatch_request
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1544, in handle_user_exception
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1639, in full_dispatch_request
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1625, in dispatch_request
  File "/vagrant/interface/views.py", line 101, in load_data
  File "/vagrant/brain/load_data.py", line 107, in load_data_append
  File "/vagrant/brain/session/data_append.py", line 76, in save_entity
  File "/vagrant/brain/database/save_entity.py", line 42, in __init__
  File "/vagrant/brain/database/db_query.py", line 39, in __init__
  File "/vagrant/log/logger.py", line 138, in __init__
  File "/usr/lib/python2.7/logging/__init__.py", line 903, in __init__
  File "/usr/lib/python2.7/logging/__init__.py", line 928, in _open
IOError: [Errno 24] Too many open files: '/vagrant/log/database/database.log'
[2016-11-27 23:38:21 +0000] [3718] [ERROR] Error handling request /load-data
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/sync.py", line 135, in handle
  File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/sync.py", line 176, in handle_request
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 2000, in __call__
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1991, in wsgi_app
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1746, in make_response
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/wrappers.py", line 847, in force_type
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/wrappers.py", line 56, in _run_wsgi_app
ImportError: No module named test
[2016-11-27 23:38:21,597] ERROR in app: Exception on /load-data [POST]
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1988, in wsgi_app
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1641, in full_dispatch_request
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1544, in handle_user_exception
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1639, in full_dispatch_request
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1625, in dispatch_request
  File "/vagrant/interface/views.py", line 103, in load_data
  File "/vagrant/brain/load_data.py", line 135, in load_model_generate
  File "/vagrant/brain/session/model_generate.py", line 37, in __init__
  File "/vagrant/brain/database/retrieve_feature.py", line 34, in __init__
  File "/vagrant/brain/database/db_query.py", line 39, in __init__
  File "/vagrant/log/logger.py", line 138, in __init__
  File "/usr/lib/python2.7/logging/__init__.py", line 903, in __init__
  File "/usr/lib/python2.7/logging/__init__.py", line 928, in _open
IOError: [Errno 24] Too many open files: '/vagrant/log/database/database.log'
[2016-11-27 23:38:21 +0000] [3718] [ERROR] Error handling request /load-data
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/sync.py", line 135, in handle
  File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/sync.py", line 176, in handle_request
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 2000, in __call__
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1991, in wsgi_app
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1746, in make_response
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/wrappers.py", line 847, in force_type
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/wrappers.py", line 56, in _run_wsgi_app
ImportError: No module named test
[2016-11-27 23:38:21,619] ERROR in app: Exception on /load-data [POST]
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1988, in wsgi_app
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1641, in full_dispatch_request
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1544, in handle_user_exception
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1639, in full_dispatch_request
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1625, in dispatch_request
  File "/vagrant/interface/views.py", line 105, in load_data
  File "/vagrant/brain/load_data.py", line 165, in load_model_predict
  File "/vagrant/brain/session/model_predict.py", line 67, in predict
  File "/vagrant/brain/database/retrieve_model_type.py", line 29, in __init__
  File "/vagrant/brain/database/db_query.py", line 39, in __init__
  File "/vagrant/log/logger.py", line 138, in __init__
  File "/usr/lib/python2.7/logging/__init__.py", line 903, in __init__
  File "/usr/lib/python2.7/logging/__init__.py", line 928, in _open
IOError: [Errno 24] Too many open files: '/vagrant/log/database/database.log'
[2016-11-27 23:38:21 +0000] [3718] [ERROR] Error handling request /load-data
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/sync.py", line 135, in handle
  File "/usr/local/lib/python2.7/dist-packages/gunicorn/workers/sync.py", line 176, in handle_request
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 2000, in __call__
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1991, in wsgi_app
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1746, in make_response
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/wrappers.py", line 847, in force_type
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/wrappers.py", line 56, in _run_wsgi_app
ImportError: No module named test
```

--- a/brain/database/db_query.py
+++ b/brain/database/db_query.py
@@ -6,7 +6,6 @@ This file contains various generic SQL-related methods.
 '''
 
 import MySQLdb as DB
-from log.logger import Logger
 from brain.database.db_settings import Database
 
 
@@ -34,9 +33,6 @@ class SQL(object):
         self.db_settings = Database()
         self.list_error = []
         self.proceed = True
-
-        # database logger
-        self.logger = Logger(__name__, 'database', 'database', 'debug')
 
         # host address
         if host:
@@ -80,10 +76,6 @@ class SQL(object):
                 )
             self.cursor = self.conn.cursor()
 
-            # log successful connection
-            if (self.cursor):
-                self.logger.log('database connected: success')
-
             return {
                 'status': True,
                 'error': None,
@@ -93,10 +85,6 @@ class SQL(object):
         except DB.Error, error:
             self.proceed = False
             self.list_error.append(error)
-
-            # log unsuccessful connection
-            if (self.cursor):
-                self.logger.log('database connected: fail')
 
             return {
                 'status': False,
@@ -125,12 +113,6 @@ class SQL(object):
                 elif sql_type == 'select':
                     result = self.cursor.fetchall()
 
-                # log transaction
-                arguments = sql_args if sql_args else 'None'
-                self.logger.log(
-                    'transaction: success, statement: %s, arguments: %s' %
-                    (sql_statement, arguments))
-
             except DB.Error, error:
                 self.conn.rollback()
                 self.list_error.append(error)
@@ -139,13 +121,6 @@ class SQL(object):
                     'error': self.list_error,
                     'result': None,
                 }
-
-                # log transaction
-                arguments = sql_args if sql_args else 'None'
-                self.logger.log(
-                    'transaction: success, statement' + sql_statement +
-                    ', arguments: ' + arguments
-                )
 
         if sql_type in ['insert', 'delete', 'update']:
             return {
@@ -173,9 +148,6 @@ class SQL(object):
                 if self.conn:
                     self.conn.close()
 
-                    # log disconnection
-                    self.logger.log('database disconnected: success')
-
                     return {
                         'status': True,
                         'error': None,
@@ -183,9 +155,6 @@ class SQL(object):
                     }
             except DB.Error, error:
                 self.list_error.append(error)
-
-                # log disconnection
-                self.logger.log('database disconnected: fail')
 
                 return {
                     'status': False,

